lib/gingr/cli.rb

```diff
@@ -111,7 +110,8 @@ def all(zipfile)
       solr(unpacked[:extract_to_path])
 
       geofile_names = unpacked[:geofile_name_hash]
-      ImportUtil.publish_geoservers(geofile_names, options)
+      geoserver_urls = options.slice(:geoserver_url, :geoserver_secure_url).transform_keys(&:to_sym)
+      Gingr::GeoserverPublisher.publish_inventory(geofile_names, **geoserver_urls)
```

Comment:
My refactorings partly aim to move logic from the ImportUtil module (which is hard to test) into the classes that make the most sense for the given code. In this case, I thought it made more sense for the publication logic to live in the GeoserverPublisher.

One issue which remains unresolved in the current codebase is that no class has simultaneous access to both the GIS files (for publication) and the metadata (which contains rights information). The original codebase addressed this with the GeoArchive and Datastore classes. That's actually a major architectural difference between the two: the old codebase acted on each datastore at a time (so you had access to the metadata and other files for a given datastore when processing it); the new codebase acts on each _filetype_ (metadata, source files, …) at a time.